dotnet/src/SemanticKernel/Memory/Storage/DataEntry.cs

```diff
@@ -33,7 +33,7 @@ public DataEntry(string key, TValue? value, DateTimeOffset? timestamp = null)
     /// Gets the key of the data.
     /// </summary>
     [JsonPropertyName("key")]
-    public readonly string Key { get; }
+    public string? Key { get; set; } = null;
```

Comment:
Does it make sense to allow the Key to be publicly set when we expect it to be set in the constructor? I'm curious on the scenario where we would modify the model after construction.